htlcswitch/link.go

```diff
@@ -2934,7 +2934,7 @@ func (l *channelLink) processExitHop(pd *lnwallet.PaymentDescriptor,
 
 	event, err := l.cfg.Registry.NotifyExitHopHtlc(
 		invoiceHash, pd.Amount, pd.Timeout, int32(heightNow),
-		circuitKey, l.hodlQueue.ChanIn(), payload,
+		circuitKey, l.hodlQueue.ChanIn(), payload, l.channel.State().ChanType.IsZeroConf(),
```

Comment:
This logic is only called for the exit-hop case, forwarding is also unsafe for the responder in a 0-conf channel flow.  I don't think you can assume that they have no other channels

Reply:
@Crypt-iQ good point. So you think we should not allow forwarding through zero-conf channels?

Reply:
@Crypt-iQ zero-conf inherently means there is trust until the channel is confirmed and the user decides to accept the risk during channel opening. In the case of received payments we allowed the option to mitigate the risk per payment by not settling automatically (holding) and in the case of forward one can mitigate it by using an interceptor.
That being said, I am opened to remove the current hold logic for the sake of consistency.

Reply:
It makes sense that the hold logic is there, I was mostly commenting because I was confused that there wasn't consistency.  You're right that zero-conf implies trust, but if it requires trust, why the holding of payments?

Reply:
Originally we thought it would be beneficial for users to have control on the amount received in zero-conf channel but I now agree that it is not necessary (definitely for the initial version of this feature). 
I will remove it.

Reply:
Done